### PR TITLE
[msbuild] Implement proper MSBuild logging in FileCopier.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Fixes https://github.com/xamarin/xamarin-macios/issues/12422.
Fixes https://github.com/xamarin/xamarin-macios/issues/13697.
Fixes https://github.com/xamarin/xamarin-macios/issues/14157.

### CODE REVIEW

File: tools/common/FileCopier.cs [hunk 1]
```diff
@@ -149,15 +194,31 @@ static CopyFileResult CopyFileCallback (CopyFileWhat what, CopyFileStep stage, I
 			}
 		}
 
+		public static bool IsUptodate (string source, string target, ReportErrorCallback reportErrorCallback, LogCallback logCallback, bool check_contents = false, bool check_stamp = true)
```
Comment (tj-devel709) [inline tools/common/FileCopier.cs:197]:
```suggestion
		public static bool IsUpToDate (string source, string target, ReportErrorCallback reportErrorCallback, LogCallback logCallback, bool check_contents = false, bool check_stamp = true)
```

Comment (tj-devel709) [reply tools/common/FileCopier.cs:197]: I think maybe this one and the ones below should have this casing, but I do see that there are already existing overloads with that casing so maybe not

**Review by tj-devel709 — APPROVED** [review]

File: tools/common/FileCopier.cs (hunk 1 above)
Comment (rolfbjarne) [reply tools/common/FileCopier.cs:197]: Yeah, I wondered about this too, but decided to not change it to not make the PR bigger than it needs to be.

Comment (mandel-macaque) [reply tools/common/FileCopier.cs:197]: If you apple that, make sure all instances are done too. I have made the mistake of using git commit suggestion to have later a build error :/

**Review by mandel-macaque — APPROVED** [review]

**Review by mauroa — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Pull request merged**